### PR TITLE
Remove ContestTopics table + Add Neynar webhook validation

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #9465
Closes: #9209

## Description of Changes
- Removes ContestTopics table in favor of ContestManager topic_id column
- Adds webhook validation for Neynar routes
- Allow null topic for farcaster contests
- Add `is_farcaster_contest` property for FC contests
- Remove FC webhook on contest cancelled/ended

## Test Plan
Set envs:
```
FLAG_FARCASTER_CONTEST=true
FLAG_WEIGHTED_TOPICS=true
```
- Run migrations
- Create a farcaster contest
- Go through "how to test" section: https://github.com/hicommonwealth/commonwealth/blob/master/common_knowledge/Farcaster-Contests.md

## Deployment Plan
N/A

## Other Considerations
I noticed that FC contests are recurring– aren't they supposed to be one-off?

### CODE REVIEW

File: libs/model/src/policies/ContestWorker.policy.ts [hunk 1]
```diff
@@ -31,9 +48,83 @@ export function ContestWorker(): Policy<typeof inputs> {
           payload.community_id,
           payload.thread_id,
         );
+
```
Comment (Rotorsoft) [inline libs/model/src/policies/ContestWorker.policy.ts:51]: should we create another query for this? similar to GetActiveConstestmanagers above?

File: libs/model/src/policies/FarcasterWorker.policy.ts [hunk 2]
```diff
@@ -110,23 +112,51 @@ export function FarcasterWorker(): Policy<typeof inputs> {
...
+            url: community.ChainNode!.private_url! || community.ChainNode!.url!,
             contest_address: contestManager.contest_address,
+            actions: [],
           },
-        });
-        mustExist('Contest Topic', contestTopic);
+        ];
 
         // create onchain content from reply cast
-        const content_url = buildFarcasterContentUrl(payload.hash);
```
Comment (Rotorsoft) [inline libs/model/src/policies/FarcasterWorker.policy.ts:121]: same as above... policies should ideally interface with queries and commands as separate artifacts, even if only for internal purposes... to promote a potentially reusable and cleaner architecture reflected in the ES model

File: packages/commonwealth/server/controllers/server_topics_methods/get_topics.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  contest_manager: schemas.ContestManager,
+});
+
+type TopicWithTotalThreads = TopicAttributes & {
+  total_threads: number;
+  active_contest_managers: Array<z.infer<typeof ActiveContestManagers>>;
+};
+
+export type GetTopicsResult = TopicWithTotalThreads[];
+export async function __getTopics(
```
Comment (Rotorsoft) [inline packages/commonwealth/server/controllers/server_topics_methods/get_topics.ts:23]: don't we have GetTopics.query already?

**Comment by Rotorsoft** [comment]
Nice job! ... just a few comments related to structuring/architecture and code quality checks should be OK before approval

**Review by masvelio — APPROVED** [review]

**Review by Rotorsoft — APPROVED** [review]
lgtm, other than the comments about not having models inside policies, but you can tackle that in the future

File: libs/model/src/policies/ContestWorker.policy.ts (hunk 1 above)
Comment (rbennettcw) [reply libs/model/src/policies/ContestWorker.policy.ts:51]: will apply on next pass

File: libs/model/src/policies/FarcasterWorker.policy.ts (hunk 2 above)
Comment (rbennettcw) [reply libs/model/src/policies/FarcasterWorker.policy.ts:121]: will apply on next pass

**Pull request merged**